src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -1073,9 +1074,21 @@ def get_extension_class(extension_name: str):
     """
     global _possible_extensions
     extensions_dict = {ext.extension_name: ext for ext in _possible_extensions}
-    assert (
-        extension_name in extensions_dict
-    ), f"Extension '{extension_name}' is not registered, please import related module before use"
+
+    if extension_name not in extensions_dict:
+        if extension_name in _builtin_extensions:
+            module = _builtin_extensions[extension_name]
+            if autoload:
+                print(f"module '{module}' is imported automatically for extension '{extension_name}'")
```

Comment:
```suggestion
```
should we remove the print?

Reply:
I think it is good to have a message for auto loading no ?
Maybe a warning

Reply:
not sure

Reply:
I'm torn here... If someone requests this functionality they should expect us to import what is needed to do the operation right? I guess the concern would be some sort of namespace clash? Like if we run an extractor the user expects a read operation from a file, we don't explicitly say we are creating a memmap which may lead to RAM exhaustion. On the other hand people complain about things failing silently for the job_kwargs.

Reply:
The extractors example is a very good point. If someone saves a JSON file with `dump_to_json`, all the modules from `extractors`, `preprocessing`, etc. will be loaded. 
In that case, we store the path to the class so imports are minimal.

Maybe we could do the same here? e.g.

```
_builtin_extensions = {
    "spike_amplitudes": "spikeinterface.postprocessing.spike_amplitudes.SpikeAmplitudeExtension",
    ...
}
```

Reply:
I think it should be removed. What is the user supposed to do with this information?

This sounds more like logging and when it is output by default the convention is to print to standard error as far as I am aware. So a warning. But I personally think that warnings should be actionable so not even sure about that one.